File: greedy/scheduling1.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

//idhere
#define forup(i, a, b)      for (int i = (a); i < (b); i++)
#define fordown(i, l, r)    for (int i = l; i >= r; i--)
#define print(a, x, y)      forup(i, x, y) cout << a[i] << " "
#define all(M)              M.begin(), M.end()
#define el                  "\n"

void fileio()
{
    #ifndef ONLINE_JUDGE
    freopen("E:/OneDrive - ptit.edu.vn/pro/dsa/input.txt", "r", stdin);
    freopen("E:/OneDrive - ptit.edu.vn/pro/dsa/output.txt", "w", stdout);
    #endif
}

bool cmp(pair<int, int> a, pair<int, int> b)
{
    return a.second < b.second;
}

int main()
{
    fileio();
    ios::sync_with_stdio(false); cin.tie(0);

    int t; cin >> t;
    while(t--)
    {
        int n; cin >> n;
        vector<pair<int, int>> events(n, {0, 0});
        forup(i, 0, n)
            cin >> events[i].first;
        forup(i, 0, n)
            cin >> events[i].second;
        sort(events.begin(), events.end(), cmp);
        for (auto x: events)
            cout << x.first << " " << x.second << el;
    }
}

File: greedy/scheduling1.py
import math, sys

#DSA03008
sys.stdin = open("E:/OneDrive - ptit.edu.vn/pro/dsa/input.txt", mode = 'r')
sys.stdout = open("E:/OneDrive - ptit.edu.vn/pro/dsa/output.txt", mode = 'w')

t = int(input())
for _ in range(t):
    n = int(input())
    start = list(map(int, input().split()))
    end = list(map(int, input().split()))
    # list with events: start and end timestamp
    all = [[start[i], end[i]] for i in range(n)]
    # sort end to ascending order
    all.sort(key = lambda x: x[1])

    res = 0
    i = 0
    while i < n - 1:
        current_end_earliest = all[i][1]
        for j in range(i + 1, n):
            start_after_current = all[j][0]
            if (current_end_earliest <= start_after_current):
                res += 1
                i = j
                break
            else:
                i += 1
    print(res + 1)